### ISSUE TITLE
Unhandled exception - datasameasexe.cfg is missing

### ISSUE THREAD

**Issue by jspittka** [contextnerror/ICC-Mac#3]
Really nice job with making this Windows app available on Mac, thanks for this!

I keep getting the following error. It doesn't affect functionality but is a bit annoying and I can't figure out why this is happening. I tried to create a file datasameasexe.cfg in drive_c/Program Files (x86)/Ice cream calculator. This fixes the problem for next time the app opens but then datasameasexe.cfg is automatically removed again and the error returns.

Any idea why this could be happening?

This is the exception that I get:

See the end of this message for details on invoking \njust-in-time (JIT) debugging instead of this dialog box.\n\n************** Exception Text **************\nSystem.IO.IOException: Unknown error (0x1700240) : 'C:\Program Files (x86)\Ice cream calculator\datasameasexe.cfg'
  at System.IO.FileSystem.DeleteFile (System.String fullPath) [0x0001a] in <e70d6e9587d64cb3abb4b3f99bbf5a0d>:0 
  at System.IO.File.Delete (System.String path) [0x00014] in <e70d6e9587d64cb3abb4b3f99bbf5a0d>:0 
  at IceCreamCalculator.Form1.A (System.Object , System.Windows.Forms.FormClosedEventArgs ) [0x00039] in <97178eee6396496380be4ccc0d1d3a0f>:0 
  at System.Windows.Forms.Form.OnFormClosed (System.Windows.Forms.FormClosedEventArgs e) [0x0002b] in <d22f8be2dd674c49bd49b314423240b8>:0 
  at System.Windows.Forms.Form.WmClose (System.Windows.Forms.Message& m) [0x00319] in <d22f8be2dd674c49bd49b314423240b8>:0 
  at System.Windows.Forms.Form.WndProc (System.Windows.Forms.Message& m) [0x001e9] in <d22f8be2dd674c49bd49b314423240b8>:0 
  at System.Windows.Forms.Control+ControlNativeWindow.OnMessage (System.Windows.Forms.Message& m) [0x00001] in <d22f8be2dd674c49bd49b314423240b8>:0 
  at System.Windows.Forms.Control+ControlNativeWindow.WndProc (System.Windows.Forms.Message& m) [0x000b3] in <d22f8be2dd674c49bd49b314423240b8>:0 
  at System.Windows.Forms.NativeWindow.Callback (System.Windows.Forms.Message& m) [0x00025] in <d22f8be2dd674c49bd49b314423240b8>:0 

\n**************

**Comment by contextnerror** [comment]
This happens when you close out the app right? Unfortunately I'm not sure why it happens either. It looks like the program wants to delete the file upon app closure, and trips over itself when it doesn't exist. But I'm not familiar enough with how the program works in an actual Windows environment to say what it's trying to accomplish by doing this, or why it's not working as it (presumably) should.

**Comment by jspittka** [comment]
Yes, it happens every time I close the app. Like I said, from what I can tell, it doesn't affect functionality but it still is a bit worrying. We can close the issue. Just thought I mention it in case somebody had a solution for this.

**Comment by contextnerror** [comment]
It's a valid issue, so I'll leave it open for visibility's sake.

**Comment by jspittka** [comment]
Just found another interesting issue related to this one. Whenever I add a new ingredient (either manually or through import from the USDA database) it disappears again after restarting the app. When I now manually add datasameasexe.cfg before adding a new ingredient they stay. Even after datasameasexe.cfg disappears in the next restart cycle the new ingredients stay. So my previous assumption that datasameasexe.cfg doesn't affect the functionality is not true any more.

**Comment by Ceolona** [comment]
I am guessing it has to do with something involving the creation of the file. Setting the app to “Place Data folder with executable" addresses the error, and appears the be the only acceptable location.
